src/frontends/tensorflow/src/graph_iterator_saved_model.hpp

```diff
@@ -85,68 +85,97 @@ class GraphIteratorSavedModel : public GraphIteratorProto {
 
+        const ::tensorflow::MetaGraphDef* meta_graph_with_no_tags = nullptr;
+
         for (const auto& meta_graph : m_saved_model->meta_graphs()) {
+            if (meta_graph.meta_info_def().tags_size() == 0) {
+                meta_graph_with_no_tags = &meta_graph;
+            }
+
             if (!meta_graph.has_graph_def()) {
                 continue;
             }
 
-            if (m_saved_model->meta_graphs_size() > 1) {
-                bool tag_found = false;
-                for (const auto& tag : meta_graph.meta_info_def().tags()) {
-                    if (tags.find(tag) != std::string::npos) {
-                        tag_found = true;
-                        break;
-                    }
+            bool tag_found = false;
+            for (const auto& tag : meta_graph.meta_info_def().tags()) {
+                if (tags.find(tag) != std::string::npos) {
+                    tag_found = true;
+                    break;
                 }
-                if (!tag_found) {
+            }
+            if (tag_found) {
+                return load_meta_graph(meta_graph);
+            }
+        }
+
+        // Alternate behavior for working with "default tag" to support additional cases for read_model
+        if (tags == "serve") {
+            // If we have only one MetaGraph - try to use it
+            if (m_saved_model->meta_graphs_size() == 1 && m_saved_model->meta_graphs(0).has_graph_def()) {
+                return load_meta_graph(m_saved_model->meta_graphs(0));
+            }
+
+            // If MetaGraph with tag == "serve" already found - we shouldn't reach this place.
+            // Otherwise we try to find a MetaGraph with no tags as an alternative
+            for (const auto& meta_graph : m_saved_model->meta_graphs()) {
+                if (!meta_graph.has_graph_def()) {
                     continue;
                 }
-            }
 
-            std::map<std::string, const ::tensorflow::SignatureDef*> validSignatures = {};
-            for (const auto& sit : meta_graph.signature_def()) {
-                const std::string& key = sit.first;
-                const ::tensorflow::SignatureDef& val = sit.second;
-                if (is_valid_signature(val)) {
-                    validSignatures[key] = &val;
+                if (meta_graph.meta_info_def().tags_size() == 0) {
+                    return load_meta_graph(meta_graph);
                 }
             }
 
-            // MetaGraph may have a list of signatures, but at this moment we need information only about
-            // "serving_default" signature which contains information about inputs/outputs names for the
-            // model. Situation when it is missing in a file also could be.
-            auto serving_default = validSignatures.find("serving_default");
+            FRONT_END_GENERAL_CHECK(false, "Saved Model doesn't contain any applicable MetaGraph");
```

Comment:
please add `[TensorFlow Frontend]`. It will help to identify TF FE error

Reply:
Np. I strongly suggest to define this part somewhere else and use it as a constant. We have nonzero-overhead adding it each time into all strings.

Reply:
We have a ticket for this activity. Agree, I also don't like it to add it everytime